src/netif.rs

```diff
@@ -369,6 +402,11 @@ impl EspNetif {
         Ok(mac)
     }
 
+    pub fn set_mac(&mut self, mac: &[u8; 6]) -> Result<(), EspError> {
+        esp!(unsafe { esp_netif_set_mac(self.1, mac.clone().as_mut_ptr() as *mut _) })?;
```

Comment:
Nit: I would assume `clone()` workarounds the fact that `esp_netif_set_mac` expects a non-const C pointer to the MAC.

I don't think the above is on purpose. In my experience, the ESP-IDF APIs don't always follow strictly C const semantics, and sometimes take non-const pointers for stuff which is not modified by these.

Or to cut the long story short `mac.as_ptr() as *const _ as *mut _` would also do, though this is such a micro-optimization that who cares after all. :)

Reply:
Thanks for the comment, I will adapt that.
The clone was mainly, because the mac is a "normal" reference and therefore `.as_mut_ptr()` was not available and it was just a quick fix for that. I will fix that.